src/Microsoft.IdentityModel.Logging/SecurityArtifact.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// An internal structure that is used to mark an argument as NonPII.
+    /// Arguments wrapped with a NonPII structure will be considered as NonPII in the message logging process.
+    /// </summary>
+    internal struct SecurityArtifact
+    {
+        /// <summary>
+        /// Argument wrapped with a <see cref="NonPII"/> structure is considered as NonPII in the message logging process.
+        /// </summary>
+        public object Argument { get; set; }
+
+        /// <summary>
+        /// Creates an instance of <see cref="NonPII"/> that wraps the <paramref name="argument"/>.
+        /// </summary>
+        /// <param name="argument">An argument that is considered as NonPII.</param>
+        public SecurityArtifact(object argument)
+        {
+            Argument = argument;
+        }
+
+        /// <summary>
+        /// Returns a string that represents the <see cref="Argument"/>.
+        /// </summary>
+        /// <returns><c>Null</c> if the <see cref="Argument"/> is <c>null</c>, otherwise calls <see cref="System.ValueType.ToString()"/> method of the <see cref="Argument"/>.</returns>
```

Comment:
supernit: we use
>  <c>null</c>
 and
> <see langword="null" />

We should standardize. I prefer the later because I think it produces better help syntax and I believe it's what .NET libraries do as well.